sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -119,7 +120,11 @@ case class AdaptiveSparkPlanExec(
       currentPhysicalPlan.execute()
     } else {
       val executionId = Option(
-        session.sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)).map(_.toLong)
+        session.sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)).flatMap { idStr =>
+        val id = idStr.toLong
+        val qe = SQLExecution.getQueryExecution(id)
+        if (qe.eq(queryExecution)) Some(id) else None
```

Comment:
Can you add some doc on why this is needed. It is kinda annoying when you have to check the git blame to figure out why code is there.